### PR TITLE
Feature hamburger menu

### PR DESCRIPTION
**Pull request by CarmenUZH**

### CODE REVIEW

**Comment by CarmenUZH** [comment]
I still have this problem where I can't see the played cards after clicking on them
Adding observer() to the GameTable solved this problem, but that was also the file that caused a merge conflict with Beat so I removed it for now

File: src/components/views/game/game/GameView.scss [hunk 1]
```diff
@@ -6,7 +6,7 @@
   flex-direction: column;
   align-items: center;
   justify-content: space-around;
-  padding: 8px 8px;
+  //padding: 8px 8px; - Removed because it caused problems with the overflow
```
Comment (BacLuc) [inline src/components/views/game/game/GameView.scss:9]: dann geht die GameView aber bis an den rand, ist auch nicht so toll

File: src/components/views/game/game/GameView.tsx [hunk 2]
```diff
@@ -47,14 +36,7 @@ const GameView = observer(() => {
...
       </Grid>
-      <Button
-        disabled={loading}
-        variant="contained"
-        endIcon={<Login />}
-        onClick={clickLeave}
-      >
-        Leave
-      </Button>
+      <HamburgerMenu />
```
Comment (BacLuc) [inline src/components/views/game/game/GameView.tsx:39]: Hätte ich jetzt am anfang erwartet (von der Reihenfolge her), und im Grid Item auf Zeile 21

File: src/components/views/game/game/yourhand/YourHand.tsx [hunk 3]
```diff
@@ -1,35 +1,34 @@
...
-    updatecards(card)
+    setTimeout(() => updatecards(card), 600)
+    console.log(toJS(activeRound.cards)) //DEBUGGING - REMOVE LATER
   }
 
   return (
     <div className="your-hand">
-      <p>These are the Cards for Match {activeMatch?.matchNumber}:</p>
+      <p style={{ marginLeft: "10px" }}>
+        These are the Cards for Match {activeMatch?.matchNumber} Round{" "}
```
Comment (BacLuc) [inline src/components/views/game/game/yourhand/YourHand.tsx:24]:
```suggestion
        These are the Cards for Match {activeMatch?.matchNumber} Round 
```
Geht das nicht auch mit einem normalen space?

File: src/components/ui/CardComponent.tsx [hunk 4]
```diff
@@ -10,7 +10,7 @@ export type CardComponentProps = {
 export const CardComponent = (props: CardComponentProps) => {
   const onClick = props.onClick ?? (() => ({}))
   return (
-    <div className="card-component" onClick={onClick}>
+    <div className="card-component ex2" onClick={onClick}>
```
Comment (BacLuc) [inline src/components/ui/CardComponent.tsx:13]: wofür steht ex2?

**Comment by BacLuc** [comment]
> Kudos, SonarCloud Quality Gate passed!    [Quality Gate passed](https://sonarcloud.io/dashboard?id=sopra-fs22-group-36_screw-your-neighbor-react&pullRequest=58)
> 
> [Bug](https://sonarcloud.io/project/issues?id=sopra-fs22-group-36_screw-your-neighbor-react&pullRequest=58&resolved=false&types=BUG) [A](https://sonarcloud.io/project/issues?id=sopra-fs22-group-36_screw-your-neighbor-react&pullRequest=58&resolved=false&types=BUG) [0 Bugs](https://sonarcloud.io/project/issues?id=sopra-fs22-group-36_screw-your-neighbor-react&pullRequest=58&resolved=false&types=BUG) [Vulnerability](https://sonarcloud.io/project/issues?id=sopra-fs22-group-36_screw-your-neighbor-react&pullRequest=58&resolved=false&types=VULNERABILITY) [A](https://sonarcloud.io/project/issues?id=sopra-fs22-group-36_screw-your-neighbor-react&pullRequest=58&resolved=false&types=VULNERABILITY) [0 Vulnerabilities](https://sonarcloud.io/project/issues?id=sopra-fs22-group-36_screw-your-neighbor-react&pullRequest=58&resolved=false&types=VULNERABILITY) [Security Hotspot](https://sonarcloud.io/project/security_hotspots?id=sopra-fs22-group-36_screw-your-neighbor-react&pullRequest=58&resolved=false&types=SECURITY_HOTSPOT) [A](https://sonarcloud.io/project/security_hotspots?id=sopra-fs22-group-36_screw-your-neighbor-react&pullRequest=58&resolved=false&types=SECURITY_HOTSPOT) [0 Security Hotspots](https://sonarcloud.io/project/security_hotspots?id=sopra-fs22-group-36_screw-your-neighbor-react&pullRequest=58&resolved=false&types=SECURITY_HOTSPOT) [Code Smell](https://sonarcloud.io/project/issues?id=sopra-fs22-group-36_screw-your-neighbor-react&pullRequest=58&resolved=false&types=CODE_SMELL) [A](https://sonarcloud.io/project/issues?id=sopra-fs22-group-36_screw-your-neighbor-react&pullRequest=58&resolved=false&types=CODE_SMELL) [1 Code Smell](https://sonarcloud.io/project/issues?id=sopra-fs22-group-36_screw-your-neighbor-react&pullRequest=58&resolved=false&types=CODE_SMELL)
> 
> [](https://sonarcloud.io/component_measures?id=sopra-fs22-group-36_screw-your-neighbor-react&pullRequest=58&metric=new_coverage&view=list) [7.1% Coverage](https://sonarcloud.io/component_measures?id=sopra-fs22-group-36_screw-your-neighbor-react&pullRequest=58&metric=new_coverage&view=list) [](https://sonarcloud.io/component_measures?id=sopra-fs22-group-36_screw-your-neighbor-react&pullRequest=58&metric=new_duplicated_lines_density&view=list) [0.0% Duplication](https://sonarcloud.io/component_measures?id=sopra-fs22-group-36_screw-your-neighbor-react&pullRequest=58&metric=new_duplicated_lines_density&view=list)

Sonar scheint nicht ganz happy zu sein

File: src/components/views/game/game/GameView.tsx [hunk 5]
```diff
@@ -1,34 +1,24 @@
...
-  const clickLeave = async () => {
-    await leaveGame()
-    navigate(Paths.LOBBY)
-  }
-
   return (
     <div className="game div-box">
       <Grid container spacing={0} className={"top-row"}>
+        <HamburgerMenu />
         <Grid item xs={1} />
```
Comment (BacLuc) [inline src/components/views/game/game/GameView.tsx:22]:
```suggestion
        <Grid item xs={1} >
            <HamburgerMenu />
        </Grid>
```
Meine Hoffnung war, dass das grid die positionierung übernehmen kann.

File: src/components/views/game/game/yourhand/YourHand.tsx [hunk 6]
```diff
@@ -1,35 +1,34 @@
...
+  const { activeMatch, yourActiveHand, activeRound } = useCurrentGame()
 
   const notYetPlayed =
     yourActiveHand?.cards.filter((value) => value.round === null) ?? []
 
   const clickCard = async (card, number) => {
-    setOpacity(number)
-    updatecards(card)
+    setTimeout(() => updatecards(card), 600)
+    console.log(toJS(activeRound.cards)) //DEBUGGING - REMOVE LATER
```
Comment (BacLuc) [inline src/components/views/game/game/yourhand/YourHand.tsx:18]:
```suggestion
```

File: src/components/views/game/game/GameView.tsx (hunk 5 above)
Comment (CarmenUZH) [reply src/components/views/game/game/GameView.tsx:22]: Ich hab mal versucht einfach das so zu machen, wie den Videochat und das Scoreboard - Weil sonst verschiebt sich alles irgendwie

File: src/components/views/game/game/GameView.tsx (hunk 2 above)
Comment (BacLuc) [reply src/components/views/game/game/GameView.tsx]: Ok, ich glaube das schauen wir uns besser synchron an

File: src/components/views/game/game/yourhand/YourHand.tsx [hunk 7]
```diff
@@ -1,35 +1,32 @@
...
-export const YourHand = () => {
+export const YourHand = observer(() => {
   const { updatecards } = useCards()
   const { activeMatch, yourActiveHand } = useCurrentGame()
-  const [opacity, setOpacity] = useState(10)
 
   const notYetPlayed =
     yourActiveHand?.cards.filter((value) => value.round === null) ?? []
 
   const clickCard = async (card, number) => {
```
Comment (BacLuc) [inline src/components/views/game/game/yourhand/YourHand.tsx:15]: sonar sagt, der number parameter is jetzt unused

**Review by BacLuc — APPROVED** [review]

**Comment by CarmenUZH** [comment]
Issue: #51

**Pull request merged**